src/daily-quest/daily-quest.validators.ts

```diff
@@ -1,19 +1,34 @@
-import { NotFoundDomainException } from "@/errors/domain";
-import { DailyQuestFormType } from "@/daily-quest/daily-quest.forms";
+import {
+  NotFoundDomainException,
+  ValidationDomainException,
+} from "@/errors/domain";
+import {
+  CreateDailyQuestFormType,
+  UpdateDailyQuestFormType,
+} from "@/daily-quest/daily-quest.forms";
 import { prisma } from "@/db/prisma";
 
-export const validateDailyQuestForUpdate = async (
-  dailyQuestId: string,
+export const validateDailyQuestForCreate = async (
   userId: string,
-  form: DailyQuestFormType
+  form: CreateDailyQuestFormType | UpdateDailyQuestFormType
 ): Promise<void> => {
   await validateGoalExists(userId, form);
-  await validateDailyQuestExists(dailyQuestId, userId);
+  await validateSuggestionExists(form);
+  await validateDailyQuestWithTitleDoesntExists(userId, form);
+};
+
+export const validateDailyQuestForUpdate = async (
+  id: string,
+  userId: string,
+  form: UpdateDailyQuestFormType
+): Promise<void> => {
+  await validateDailyQuestExists(id, userId);
+  await validateDailyQuestWithTitleDoesntExists(userId, form);
```

Comment:
```suggestion
  await validateDailyQuestTitleDoesNotExist(userId, form);
```

Reply:
This validator checks if DailyQuest with the same title for user exists, not the DailyQuest title itself.

Reply:
Fixed!